cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1374,22 +1387,99 @@ private KafkaVersionChange getKafkaVersionChange(StatefulSet kafkaSts) {
+                                        "update process defined in the documentation should be followed. When finished set " +
+                                         "%s annotation on the Zookeeper StatefulSet to false.",
+                                        Annotations.ANNO_STRIMZI_IO_MANUAL_ZK_SCALE);
+
+                                log.info(scaleMessage);
 
+                                zkManualScalingCondition = new ConditionBuilder()
+                                        .withLastTransitionTime(new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ").format(dateSupplier()))
+                                        .withType("ZK-Manual-Scaling")
+                                        .withReason("replicaChange")
+                                        .withStatus("True")
+                                        .withMessage(scaleMessage);
+
+                                // Setting this will add the annotation to the STS in the state.zkStatefulSet() method
+                                zkManualScaling = true;
+
+                            } else if (Annotations.booleanAnnotation(sts, Annotations.ANNO_STRIMZI_IO_MANUAL_ZK_SCALE, false)) {
+                                // This is an on-going manual scaling operation. Once the annotation has been set to true by the operator
+                                // only the user can set it to false.
+                                String scaleMessage = String.format("Detected manual Zookeeper scaling operation in progress. " +
+                                        "Rolling of Zookeeper cluster will remain disabled until %s annotation is set to false.",
+                                        Annotations.ANNO_STRIMZI_IO_MANUAL_ZK_SCALE);
+
+                                log.info(scaleMessage);
+
+                                zkManualScalingCondition = new ConditionBuilder()
+                                        .withLastTransitionTime(new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ").format(dateSupplier()))
+                                        .withType("ZK-Manual-Scaling")
+                                        .withReason("manualScaleAnnotationSet")
+                                        .withStatus("True")
+                                        .withMessage(scaleMessage);
+
+                                zkManualScaling = true;
+
+                            } else {
+                                // Either this is a ZK 3.4.x cluster or the user is not manually scaling the 3.5.x cluster.
+                                zkManualScaling = false;
+                            }
+
+                            if (zkManualScalingCondition != null && isZK35x) {
+                                List<Condition> oldConditions = this.kafkaStatus.getConditions();
+                                List<Condition> newConditions = oldConditions != null ? new ArrayList<>(oldConditions) : new ArrayList<>();
```

Comment:
Again, what is the advantage of always creating new ArrayList over using the old one if it already exists? Also, this code seems to be used now at least twice - maybe we can factor it out?

Reply:
So as above this is due to the Status object keeping the Conditions in an unmodifiable list. Should we add a method to the abstract Status class that allows adding conditions to its internal list. That would remove the need for repeating this code.

Reply:
Yeah, I gues either there or in some Utils. Why is it using unmodifiable list? And is it ok if you replace it with regular ArrayList then? Should it be also unmodifiable?